src/reducer.js

```diff
@@ -0,0 +1,58 @@
+import { v4 as uuidv4 } from 'uuid';
+
+export const TODO_DONE_TYPE = 'TASK_DONE';
+export const ADD_TODO_TYPE = 'ADD_TODO';
+export const TODO_REMOVE_TYPE = 'TASK_REMOVE';
+export const RESET_TYPE = 'SET_UP';
+export const REMOVE_COMPLETED_TYPE = 'REMOVE_COMPLETED';
+
+const KEY_TODO = 'KEY_TODO';
+
+const reset = () => JSON.parse(localStorage.getItem(KEY_TODO));
+
+const todoDone = (todos, id, done) => {
+  const newTodos = todos.map(obj => {
+    if (obj.id === id) {
+      return { ...obj, done: done };
+    }
+    return obj;
+  });
+  localStorage.setItem(KEY_TODO, JSON.stringify(newTodos));
+  return newTodos;
+};
+
+const todoRemove = (todos, id) => {
+  const newTodos = todos.filter(x => x.id !== id).slice();
+  localStorage.setItem(KEY_TODO, JSON.stringify(newTodos));
+  return newTodos;
+};
+
+const removeCompleted = todos => {
+  const newTodos = todos.filter(todo => !todo.done);
+  localStorage.setItem(KEY_TODO, JSON.stringify(newTodos));
+  return newTodos;
+};
+
+const addTodo = (todos, value) => {
+  const todo = { title: value, done: false, id: uuidv4() };
+  const newDotos = todos ? [...todos.slice(), todo] : [todo];
```

Comment:
Slice is redundant in this case. Actually, you can set default value as an empty array.